### PR TITLE
[Example] 新增 A5 HISA Paged Block-Sparse MQA Decode 算子示例

### PR DESCRIPTION
**Pull request by ShareableXue**
# feat(HISA): 新增 A5 persistent paged block-sparse MQA decode 算子

## 背景

现有 paged block-sparse MQA decode 示例面向 A3，score 的搬运路径为：

```text
L0C -> GM workspace -> AIV UB
```

A5 增加了 FixPipe 到 AIV UB 的直接通路，并且 TileLang Ascend 在 A5 上使用 PTO-ISA
后端。若只机械替换 score 搬运路径，多 batch 下仍会受到 one-task-per-block mixed
block 生命周期和跨 wave 调度空洞的影响，B2/B4/B8 无法充分重叠 AIC 与 AIV。

本 PR 新增 A5 专用实现，在保留 A3 四块计算骨架和算子数学语义的前提下，使用
定向 C2V 搬运和统一 persistent 调度。

### 1. 使用 A5 定向 FixPipe -> AIV UB 通路

- 编译目标固定为 `target="pto", platform="A5"`；
- 用四次 `copy_cv_experiment(..., SingleVec0/SingleVec1)` 将四个 L0C score tile
  分别写入目标 AIV 的 early/late UB；
- 删除 fp32 score GM workspace、workspace kernel 参数和 workspace -> UB DMA；
- 用定向 ready event 通知唯一目标 AIV，避免向非目标 AIV 广播后留下未消费 token；
- 显式关闭会重写手工 C2V 和同步拓扑的 PTO pass。

默认 `topk=64, heads=32, kv_block_size=128` 时，每个 token 的有效 score 为 1 MiB。
新路径删除约 1 MiB L0C -> GM 写和约 1 MiB GM -> UB 读。

### 2. 统一使用 persistent physical grid

logical task 定义不变，每个 task 仍处理连续四个 top-k block：

```text
logical_grid  = total_tokens * (topk / 4)
physical_grid = min(logical_grid, aicore_num)
```

resident mixed block 通过 `T.Persistent` 连续领取 logical task，不再为每个 logical
task单独建立和销毁 mixed block。AIC 可以生产下一 task，两个 AIV 同时消费当前
task，从而减少多 wave 下的调度空洞。

所有 batch 使用同一套调度。B1 只有一轮迭代，PTO 会将其静态折叠为单 slot，不存在
单独的 B1 运行时分支。

### 3. 双 score slot 完整复用

- 多 wave 按 `score_slot = (task // physical_grid) % 2` 交替使用两套 early/late
  score UB；
- slot0 ready event 为 `0/17/2/19`，slot1 为 `4/21/6/23`；
- 第三次覆盖同一 slot 前，AIV 在 late reduction 后发布 release 8/9，AIC 在下一次
  TMOV 前等待 release；
- 进入下一 persistent task 并覆盖 Q L0A 前，同时等待两个 L0B ping-pong 的
  `M -> MTE1` token，保证上一 task 的 MMA2/MMA3 已停止读取 L0A。

双槽仅复制 score UB；logits、weight 和 reduction 临时区仍按 task 复用。默认 shape
下每个 AIV 的 persistent UB 约为 68,800 B，低于 9579 的 248 KiB UB 上限。

### 4. 优化 ContextLens mask 并补齐 Scalar 同步

mask 的数学定义不变，按块拆成：

- full：整块有效，不执行 mask；
- invalid：整块无效，Vector fill `-inf`；
- partial：Scalar 只写 `[valid, kv_block_size)`。

partial 分支显式执行：

```text
reduce_sum(V) -> V -> S -> Scalar SetValue -> S -> MTE3 -> output
```

该依赖防止 Scalar 尾部写入被尚未完成的 reduction 覆盖，也防止 MTE3 在尾部写入
可见前读取输出。full 热路径没有 mask 或额外 Scalar 同步。

### 5. 保留可直接运行的 standalone 接口

该算子文件新增 `--aicore_num` 用于目标 SKU 特化。

## 性能结果

环境：Ascend950PR_9579，28 AIC / 56 AIV。task 与 persistent 调度各独立采集5次，
单位为微秒：

| batch | task P50 | persistent P50 | P50 降幅 | task P90 | persistent P90 |
|---:|---:|---:|---:|---:|---:|
| 1 | 7.926 | 7.898 | 0.35% | 8.056 | 8.038 |
| 2 | 13.439 | 12.015 | 10.60% | 13.625 | 12.051 |
| 4 | 23.489 | 13.044 | 44.47% | 23.699 | 14.140 |
| 8 | 45.333 | 15.318 | 66.21% | 46.383 | 15.484 |

（该数据来自开发过程中的非标准不严谨测试，最终性能输出并不以此为标准，需要实机多次测量，实际可能略逊于以上）
B8 对比中，两种调度的总 Vector active time 约为 159.47/159.83 us，GM -> L1 数据量
约为 17.52/17.48 MiB；persistent 并没有少算或少读 KV，收益来自 resident C/V 对跨
wave 调度空洞的消除。persistent B8 的关键核最大 AIC/AIV 时间约为
14.55/14.15 us，二者已基本均衡。

## 验证结果

- A5 精度全部通过；
- B4 mixed-tail 连续100次 launch 通过；
- B8 连续100次 launch 通过；
- B1 统一 persistent 与 task 调度的性能差异在测试误差内；
- Python 语法检查通过。

## 兼容性和使用约束

- 本实现是 A5 PTO-ISA 专用示例，不替换 A3 实现；
- 设计目标是 decode 场景，常用配置为 `seq_len=1`；
- `topk` 必须是4的倍数；
- 默认 `aicore_num=28` 对应 Ascend950PR_95XX；
- Ascend950DT_95XX 为36个 AIC，需传入 `--aicore_num 36` 并在该 SKU 上重新执行
  精度、资源和性能门禁；
- 不同 SKU 的逻辑计算和同步协议不变。

## 提交文件

- `examples/HISA/paged_block_sparse_mqa_attn_a5_expert.py`

### CODE REVIEW

**Review by LLMZhangYC — DISMISSED** [review]
lgtm

**Review by LLMZhangYC — APPROVED** [review]
lgtm
